network-libp2p/src/rate_limiting.rs

```diff
@@ -141,114 +163,120 @@ impl RateLimit {
 }
 
 // Network helpers for rate limiting
+#[derive(Default)]
+pub(crate) struct RateLimits {
+    peer_request_limits: HashMap<PeerId, HashMap<RequestType, RateLimit>>,
+    rate_limits_pending_deletion: PendingDeletion,
+}
 
-pub(crate) fn is_under_the_rate_limits<Req: RequestCommon>(
-    peer_request_limits: Arc<Mutex<HashMap<PeerId, HashMap<u16, RateLimit>>>>,
-    peer_id: PeerId,
-    request_id: InboundRequestId,
-) -> bool {
-    // Gets lock of peer requests limits read and write on it.
-    let mut peer_request_limits = peer_request_limits.lock();
-
-    // If the peer has never sent a request of this type, creates a new entry.
-    let requests_limit = peer_request_limits
-        .entry(peer_id)
-        .or_default()
-        .entry(Req::TYPE_ID)
-        .or_insert_with(|| RateLimit::new(Req::MAX_REQUESTS, Req::TIME_WINDOW, Instant::now()));
-
-    // Ensures that the request is allowed based on the set limits and updates the counter.
-    // Returns early if not allowed.
-    if !requests_limit.increment_and_is_allowed(1) {
-        log::debug!(
-            "[{:?}][{:?}] {:?} Exceeded max requests rate {:?} requests per {:?} seconds",
-            request_id,
-            peer_id,
-            std::any::type_name::<Req>(),
-            Req::MAX_REQUESTS,
-            Req::TIME_WINDOW,
-        );
-        return false;
+impl RateLimits {
+    pub(crate) fn exceeds_rate_limit(
+        &mut self,
+        peer_id: PeerId,
+        request_type: RequestType,
+        request_id: InboundRequestId,
+        request_rate_limit_data: &RequestRateLimitData,
+    ) -> bool {
+        // If the peer has never sent a request of this type, creates a new entry.
+        let requests_limit = self
+            .peer_request_limits
+            .entry(peer_id)
+            .or_default()
+            .entry(request_type)
+            .or_insert_with(|| {
+                RateLimit::new(
+                    request_rate_limit_data.max_requests,
+                    request_rate_limit_data.time_window,
+                    Instant::now(),
+                )
+            });
+
+        // Ensures that the request is allowed based on the set limits and updates the counter.
+        // Returns early if not allowed.
+        if !requests_limit.increment_and_is_allowed(1) {
+            info!(
+                %request_id,
+                %peer_id,
+                %request_type,
+                "Rate limit was exceeded!",
+            );
+            log::debug!(
+                "[{:?}][{:?}] {:?} Exceeded max requests rate {:?} requests per {:?} seconds",
+                request_id,
+                peer_id,
+                request_type,
+                request_rate_limit_data.max_requests,
+                request_rate_limit_data.time_window,
+            );
+            return true;
+        }
+
+        false
```

Comment:
I suggest keeping the debug log only. it is more informativ. Also I'd suggest to use tracing notation. Also if you log this callerside this can be replaced by 
```suggestion
        requests_limit.increment_and_is_allowed(1)
```